addons/l10n_sa_edi/data/ubl_21_zatca.xml

```diff
@@ -122,53 +122,83 @@
         </template>
 
         <template id="ubl_21_InvoiceLineType_zatca" inherit_id="account_edi_ubl_cii.ubl_20_InvoiceLineType" primary="True">
-            <!-- Remove SellersItemIdentification if None -->
-            <xpath expr="//*[local-name()='SellersItemIdentification']" position="replace">
-                <cac:SellersItemIdentification t-if="item_vals['sellers_item_identification_vals']['id']"
-                                               xmlns="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2"
-                                               xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2"
-                                               xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2">
-                    <cbc:ID t-out="item_vals['sellers_item_identification_vals']['id']"/>
-                </cac:SellersItemIdentification>
+            <xpath expr="//*[local-name()='PricingReference']" position="before">
+                <t xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2"
+                   xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2">
+                    <cac:DocumentReference>
+                        <t t-set="prepayment_vals" t-value="vals.get('prepayment_vals', {})"/>
+                        <t t-set="issue_date" t-value="prepayment_vals.get('issue_date')"/>
+                        <cbc:ID t-out="prepayment_vals.get('prepayment_id')"/>
+                        <cbc:IssueDate t-out="issue_date and issue_date.strftime('%Y-%m-%d')"/>
+                        <cbc:IssueTime t-esc="issue_date and issue_date.strftime('%H:%M:%S')"/>
+                        <cbc:DocumentTypeCode t-out="prepayment_vals.get('document_type_code')"/>
+                    </cac:DocumentReference>
```

Comment:
can't we add it in the UBL 2.0 Common template ? It's used by the CreditNoteLine, DebitNoteLine, and InvoiceLine (http://www.datypic.com/sc/ubl20/e-cac_DocumentReference.html).